File: taurus-protect-sdk-go/pkg/protect/model/currency.go
package model

// Currency represents a cryptocurrency or fiat currency.
type Currency struct {
	// ID is the unique identifier of the currency.
	ID string `json:"id"`
	// Name is the name of the currency (e.g., "ethereum").
	Name string `json:"name"`
	// Symbol is the shorthand symbol for the currency (e.g., "ETH").
	Symbol string `json:"symbol"`
	// DisplayName is the display name for the currency (e.g., "Ethereum").
	DisplayName string `json:"display_name"`
	// Type is the type of currency: "token", "fiat", "native", or "signet".
	Type string `json:"type"`
	// Blockchain is the blockchain the currency is associated with (e.g., "ETH", "BTC").
	Blockchain string `json:"blockchain"`
	// Network is the network or environment (e.g., "mainnet", "testnet").
	Network string `json:"network"`
	// Decimals is the number of decimal places the currency uses (e.g., 18 for ETH).
	Decimals int64 `json:"decimals"`
	// CoinTypeIndex is the BIP44 coin type index (e.g., "0" for Bitcoin, "60" for Ethereum).
	CoinTypeIndex string `json:"coin_type_index"`
	// ContractAddress is the smart contract address if currency is a token.
	ContractAddress string `json:"contract_address,omitempty"`
	// TokenID is the unique identifier for the token (used for NFTs and some blockchains).
	TokenID string `json:"token_id,omitempty"`
	// WlcaID is the whitelisted contract address ID associated with the currency.
	WlcaID int64 `json:"wlca_id,omitempty"`
	// Logo is the currency logo in Data URI scheme (base64 encoded).
	Logo string `json:"logo,omitempty"`
	// IsToken indicates if the currency is a token.
	IsToken bool `json:"is_token"`
	// IsERC20 indicates if the token is an ERC-20 token.
	IsERC20 bool `json:"is_erc20"`
	// IsFA12 indicates if the currency is based on FA12 standard (Tezos).
	IsFA12 bool `json:"is_fa12"`
	// IsFA20 indicates if the currency is based on FA20 standard (Tezos).
	IsFA20 bool `json:"is_fa20"`
	// IsNFT indicates if the currency represents a Non-Fungible Token.
	IsNFT bool `json:"is_nft"`
	// IsFiat indicates if the currency is a fiat currency (e.g., CHF, EUR, USD).
	IsFiat bool `json:"is_fiat"`
	// IsUTXOBased indicates if the currency is UTXO-based (e.g., Bitcoin).
	IsUTXOBased bool `json:"is_utxo_based"`
	// IsAccountBased indicates if the currency is account-based (e.g., Ethereum).
	IsAccountBased bool `json:"is_account_based"`
	// HasStaking indicates if the currency supports staking.
	HasStaking bool `json:"has_staking"`
	// Enabled indicates if the currency is enabled in the current tenant.
	Enabled bool `json:"enabled"`
}

// ListCurrenciesOptions contains options for listing currencies.
type ListCurrenciesOptions struct {
	// ShowDisabled includes currencies disabled by business rules when true.
	ShowDisabled bool
	// IncludeLogo includes currency logos in the response when true.
	IncludeLogo bool
}

// GetCurrencyOptions contains options for retrieving a single currency.
type GetCurrencyOptions struct {
	// Blockchain is the blockchain to filter by (required, e.g., "ETH", "BTC").
	Blockchain string
	// Network is the network or environment (required, e.g., "mainnet", "testnet").
	Network string
	// CurrencyID is the unique identifier of the currency to filter by (optional).
	CurrencyID string
	// TokenContractAddress filters by token contract address (optional).
	// If set, returns a token with the specified contract address.
	// If not set, returns the native currency of the blockchain.
	TokenContractAddress string
	// TokenID differentiates assets within a contract address (optional).
	// Used for blockchains like ALGO and XTZ where one contract may contain multiple assets.
	TokenID string
	// ShowDisabled includes the currency even if disabled by business rules when true.
	ShowDisabled bool
	// IncludeLogo includes the currency logo in the response when true.
	IncludeLogo bool
}

File: taurus-protect-sdk-typescript/src/models/currency.ts
/**
 * Currency models for Taurus-PROTECT SDK.
 */

/**
 * Currency information.
 */
export interface Currency {
  /** Unique currency identifier */
  readonly id?: string;
  /** Currency symbol (e.g., "ETH", "BTC") */
  readonly symbol?: string;
  /** Currency name */
  readonly name?: string;
  /** Display name for UI presentation */
  readonly displayName?: string;
  /** Currency type */
  readonly type?: string;
  /** Blockchain type */
  readonly blockchain?: string;
  /** Network (e.g., "mainnet", "testnet") */
  readonly network?: string;
  /** Number of decimal places */
  readonly decimals?: number;
  /** BIP-44 coin type index */
  readonly coinTypeIndex?: string;
  /** Token contract address (for tokens) */
  readonly contractAddress?: string;
  /** Token ID (for multi-asset contracts) */
  readonly tokenId?: string;
  /** WLCA identifier */
  readonly wlcaId?: number;
  /** Logo URL or path */
  readonly logo?: string;
  /** Whether this is a token (not the native currency) */
  readonly isToken: boolean;
  /** Whether this is an ERC-20 token */
  readonly isERC20: boolean;
  /** Whether this is an FA1.2 token (Tezos) */
  readonly isFA12: boolean;
  /** Whether this is an FA2.0 token (Tezos) */
  readonly isFA20: boolean;
  /** Whether this is an NFT */
  readonly isNFT: boolean;
  /** Whether this currency uses UTXO-based transactions */
  readonly isUTXOBased: boolean;
  /** Whether this currency uses account-based transactions */
  readonly isAccountBased: boolean;
  /** Whether this is a fiat currency */
  readonly isFiat: boolean;
  /** Whether this currency supports staking */
  readonly hasStaking: boolean;
  /** Whether this currency is enabled */
  readonly enabled: boolean;
  /** Whether this is the native currency of the blockchain (backward compat) */
  readonly isNative?: boolean;
  /** Whether this currency is disabled (backward compat) */
  readonly isDisabled?: boolean;
  /** Logo URL (backward compat, prefer logo) */
  readonly logoUrl?: string;
  /** Price in base currency (backward compat) */
  readonly price?: string;
  /** Base currency used for pricing (backward compat) */
  readonly priceCurrency?: string;
}

/**
 * Options for listing currencies.
 */
export interface ListCurrenciesOptions {
  /** Include disabled currencies */
  showDisabled?: boolean;
  /** Include logo URLs in response */
  includeLogo?: boolean;
}

/**
 * Options for getting a currency by blockchain.
 */
export interface GetCurrencyByBlockchainOptions {
  /** Blockchain type (required) */
  blockchain: string;
  /** Network (required) */
  network: string;
  /** Token contract address (optional, returns native currency if not specified) */
  contractAddress?: string;
  /** Token ID (optional, for multi-asset contracts) */
  tokenId?: string;
}
